src/ListBuilder.php

```diff
@@ -390,24 +413,36 @@ public function buildPagerInfo(ContentEntityInterface $entity): array {
    *   The facet.
    * @param string $value
    *   The raw value.
+   * @param \Drupal\oe_list_pages\ListSourceInterface $list_source
+   *   The list source.
    *
    * @return string|null
    *   The display value.
    */
-  protected function getFacetResultDisplayLabel(FacetInterface $facet, string $value): ?string {
+  protected function getFacetResultDisplayLabel(FacetInterface $facet, string $value, ListSourceInterface $list_source): ?string {
     if ($facet->getWidgetInstance() instanceof FulltextWidget) {
       // For facets that use the full text widget, the actual value is the
       // selected item.
       return $value;
     }
 
-    foreach ($facet->getResults() as $facet_result) {
-      if ($facet_result->getRawValue() === $value) {
-        return (string) $facet_result->getDisplayValue();
-      }
+    $field_definition = $this->getFacetFieldDefinition($facet, $list_source);
```

Comment:
There is a problem here when the indexed field does not map to an actual drupal field (like call_proposals_last_deadline). In this scenario we don't get a definition so we will get an error.